src/kademlia/node.cpp

```diff
@@ -115,9 +98,16 @@ node::node(udp_socket_interface* sock
 	, m_last_self_refresh(min_time())
 	, m_sock(sock)
 	, m_counters(cnt)
+	, m_storage(dht_default_storage_constructor(m_id, m_settings, m_counters))
 {
 	m_secret[0] = random();
 	m_secret[1] = random();
+
+	TORRENT_ASSERT(m_storage != NULL);
+}
+
+node::~node() {
+	if (m_storage != NULL) delete m_storage;
```

Comment:
in this case I don't believe node is copyable/movable or assignable. but in the general case, having raw pointers as members would likely cause problems with those operations and are code smells